os/fs/driver/mtd/mtd_config.c

```diff
@@ -1160,21 +1160,19 @@ static int mtdconfig_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 	FAR struct mtdconfig_struct_s *dev = inode->i_private;
 	FAR struct config_data_s *pdata;
 	int ret = -ENOSYS;
+	pdata = (FAR struct config_data_s *)arg;
+	if(!pdata) {
```

Comment:
1. It looks little strange.
Your code is `put data and then check validation`.
Logically, `put data after check validation` looks better.
If it is invalid, we don't need to re-assign to `pdata`.
2. Returning ENOSYS is not approprite. EINVAL is better.
```
if (!arg) {
    return -EINVAL;
}
pdata = (....)arg;
```

Reply:
thanks for your suggestion, i agree and will fix it.